libs/core/scripts/build-tokens.ts

```diff
@@ -1,32 +1,99 @@
 import fs from 'fs-extra';
 import tokens from '../tokens/tokens.json';
 
+// This is specific to ordering short-hand
+// CSS properties that require a specific order
+const propertyOrder = {
+	"border": ['width', 'style', 'color'],
+	"box-shadow": ['x', 'y', 'blur', 'spread', 'color']
+};
+
 /**
- * Takes the JSON data and converts it to formatted CSS
- * custom properties.
- * 
- * @param tokens Tokens data in JSON format.
- * @param base Base string to format custom properties.
- * @returns Formatted CSS with all custom properties.
+ * Runs the process to build tokens from JSON and save them to their
+ * appropriate files.
+ *
+ * @param data Tokens data in JSON format
  */
-const tokensToCss = (tokens: object, base = `-`) =>
-  Object.entries(tokens).reduce((css, [key, value]) => {
-		let newBase = base + `-${key}`;
-		if (!['description', 'type'].includes(key)) {
-			newBase = newBase.replace(/\-value/, '');
-			if (typeof value !== "object") {
-				return css + newBase + `: ${value};\n`
-			}
-			return css + tokensToCss(value, newBase)
-		} 
-		return css
-  }, ``);
+const build = (data: object) => {
+
+	try {
+		Object.keys(data).map((cssPropertyName) => {
+			const cssVariables = tokensToCss(tokens[cssPropertyName], cssPropertyName, `--${cssPropertyName}`, ).replaceAll('', '');
+			const cssClass = `:root {\n${cssVariables.replaceAll("--", "  --sage-")}}\n`
+			saveTokens(cssPropertyName, cssClass)
+		})
+
+	} catch (e) {
+		console.log(
+			"Provide a correct argument - a relative path to design tokens.\n",
+			e
+		)
+	}
+}
+
+build(tokens);
```

Comment:
You were right: looks like moving this line up to kick off the build process is the reason for the error I'm seeing. I'd suggest moving it back to the end of the file, along with the entire `build` function block to keep it all together.

```
Provide a correct argument - a relative path to design tokens.
 ReferenceError: Cannot access 'tokensToCss' before initialization
    at /Users/kevin.chang/Sites/sage/libs/core/scripts/build-tokens.js:34:34
```

Reply:
Just as a heads up @monkeypox8, we would like to keep our functions/methods in alphabetical order. This works since they have to be compiled into javascript in most cases.

Reply:
Nice, I'm a fan of this as a convention. My only concern would be with synchronous functions similar to here, but in all likelihood they'll be pretty rare.